client/src/index.jsx

```diff
@@ -11,29 +12,28 @@ class App extends React.Component {
       product_id: null
     };
     this.singleItemRequest = this.singleItemRequest.bind(this);
+    this.getStartingPid = this.getStartingPid.bind(this);
   }
 
   componentDidMount() {
-    getStartingPid()
-    .then((id) => {
-      this.setState({
-        product_id: id
-      })
-    })
-    .catch((error) => {
-      console.log(error);
-    })
-
-    //example usage for SIR
-    //this.singleItemRequest(38322);
+    this.singleItemRequest();
   }
 
   singleItemRequest (id) {
-    var config = {
-      method:'get',
-      url:`http://localhost:10038/singleItemRequest`,
-      params: {product_id: id}
-    };
+
+    var config = {}
+    if (id) {
+      config = {
```

Comment:
oo this would look great with a ternary

```node.js
const CONFIG = id
  ? {
       method: "get",
       url: "http://localhost:10038/singleItemRequest",
       params: { product_id: id },
     }
  : {
       method: "get",
       url: "http://localhost:10038/starting_product_id",
    }
```